components/Reformeur.jsx

```diff
@@ -227,26 +227,30 @@ class Reformeur extends Component {
 
     UpdateBareme = (i, value) => {
         const ref = this.state.reforme
-        const list = this.state.reforme.impot_revenu.bareme.seuils.map((item, j) => {
-            if (j === i) {
-                const valchiffre = parseInt(value, 10)
-                return isNaN(valchiffre) ? item : valchiffre
-            }
-            return item
-        })
+        const list = this.state.reforme.impot_revenu.bareme.seuils.map(
+            (item, j) => {
+                if (j === i) {
+                    const valchiffre = parseInt(value, 10)
+                    return isNaN(valchiffre) ? item : valchiffre
+                }
+                return item
+            },
+        )
```

Comment:
Les choses que je vois ici:

- le chainage des properties `this.state.reforme.impot_revenu.bareme.seuil` lodash.get peut faire le taff plus efficacement, il est capable de memoization et permet d'éviter les fallbacks en cas de props non-définies dans la chaine

- le map qui peut être écrit en fonctionnel en exportant la function > plus lisible/testable
```
function doSomething(item, j) {
  ...
}
this.state.reforme.impot_revenu.bareme.seuils.map(doSomething)
```

- le `if` peut être inversé pour éviter le bruit dans la lecture par l'indentation, avec une variable pour expliciter la condition
```
const isCurrentIndex = j === i
if (!isCurrentIndex) return item
const valchiffre = parseInt(value, 10)
return isNaN(valchiffre) ? item : valchiffre
```